src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -1193,20 +1193,32 @@ AS
 
     BEGIN TRANSACTION
         
-        -- Check if tag exsit
+        -- Lock the tag from external updates
         DECLARE @tagStatus TINYINT
         DECLARE @tagKey BIGINT
         SELECT @tagKey = TagKey, @tagStatus = TagStatus
         FROM dbo.CustomTag WITH(HOLDLOCK) 
         WHERE dbo.CustomTag.TagPath = @tagPath
 
+        -- Check existence
         IF @@ROWCOUNT = 0
             THROW 50404, 'custom tag not found', 1 
 
         -- check if status is Added
         IF @tagStatus <> 1 
             THROW 50412, 'custom tag is not in status Added', 1
 
+        -- Update status to Deindexing
+        UPDATE dbo.CustomTag SET TagStatus = 2 
+        WHERE dbo.CustomTag.TagKey = @tagKey
+
+    COMMIT TRANSACTION
+
+    BEGIN TRANSACTION
+        -- Lock the tag from external updates
+        SELECT TagKey FROM dbo.CustomTag WITH(HOLDLOCK)
+        WHERE dbo.CustomTag.TagKey = @tagKey
```

Comment:
Can anything update a custom tag with status 2?

Reply:
DeleteCustomTag is the only SP that could update to status 2, but I'm thinking it might be safer to check the status at the second transaction.

Reply:
Sure, checking makes sense - I'm more interested in the lock you are putting on the row!

Reply:
I'm also unsure of if this solves the problem - what if someone retrieves custom tags prior to the status update? Will STOW try to update the custom tag status to 1 and be blocked by the lock?

Reply:
yes, STOW should lock that custom tag at first preventing changes from DeleteCustomTag.

Reply:
Can you run a quick check on 2 query window with WAITFORDELAY. 

My assumption is once a procedure does a select a shared lock is held. That means your select which needs update lock cannot get the exclusive lock untill STOW procedure is done.

We should avoid locks in STOW, this will make the stored procedures completely serial